### PR TITLE
[Docs] Convert examples to mdoc

### PR DESCRIPTION
**Pull request by ghostbuster91**
- [x] community.md
- [x] examples.md
- [x] goals.md
- [x] how.md
- [x] index.md
- [x] json.md
- [x] other.md
- [x] quickstart.md
- [x] resilience.md
- [x] testing.md
- [x] websockets.md
- [x] 0001-extract-stream-from-http4s.md
- [x] 0002-http-model-conventions.md
- [x] akka.md
- [x] catseffect.md
- [ ] finagle.md
- [x] fs2.md
- [x] future.md
- [x] http4s.md
- [x] monix.md
- [x] scalaz.md
- [x] start_stop.md
- [x] summary.md
- [x] synchronous.md
- [x] zio.md
- [ ] fetch.md
- [ ] curl.md
- [x] custom.md
- [x] opentracing.md
- [x] prometheus.md
- [x] slf4j.md
- [x] zio-opentracing.md
- [x] proxy.md
- [x] redirects.md
- [x] ssl.md
- [x] timeouts.md
- [x] model.md
- [x] uri.md
- [x] authentication.md
- [x] basics.md
- [x] body.md
- [x] cookies.md
- [x] headers.md
- [x] multipart.md
- [x] streaming.md
- [x] type.md
- [x] basics.md
- [x] body.md
- [x] exceptions.md

Before submitting pull request:
- [ ] Check if the project compiles by running `sbt compile`
- [ ] Verify docs compilation by running `sbt compileDocs`
- [ ] Check if tests pass by running `sbt test`
- [ ] Format code by running `sbt scalafmt`

### CODE REVIEW

File: docs/backends/catseffect.md [hunk 1]
```diff
@@ -7,6 +7,15 @@ To use, add the following dependency to your project:
 ```scala
 "com.softwaremill.sttp.client" %% "async-http-client-backend-cats" % "2.2.0"
 ```
+And some imports:
```
Comment (adamw) [inline docs/backends/catseffect.md:10]:
```suggestion
You'll need the following imports and implicits to create the backend:
```

File: docs/backends/catseffect.md [hunk 2]
```diff
@@ -7,6 +7,15 @@ To use, add the following dependency to your project:
...
+```scala mdoc:silent
+import sttp.client._
+import sttp.client.asynchttpclient.cats.AsyncHttpClientCatsBackend
+import cats.effect._
+
+// an implicit ContextShift in required to create the backend; here, for `cats.effect.IO`:
+implicit val cs: ContextShift[IO] = IO.contextShift( scala.concurrent.ExecutionContext.global )
```
Comment (adamw) [inline docs/backends/catseffect.md:17]:
```suggestion
implicit val cs: ContextShift[IO] = IO.contextShift(scala.concurrent.ExecutionContext.global)
```

File: docs/backends/fs2.md [hunk 3]
```diff
@@ -9,6 +9,15 @@ To use, add the following dependency to your project:
...
 "com.softwaremill.sttp.client" %% "async-http-client-backend-fs2" % "2.2.0"
 ```
+And some imports:
+```scala mdoc:silent
+import sttp.client.asynchttpclient.fs2.AsyncHttpClientFs2Backend
+import cats.effect._
+import sttp.client._
+
+// an implicit ContextShift in required to create a concurrent instance for `cats.effect.IO`:
+implicit val cs: ContextShift[IO] = IO.contextShift( scala.concurrent.ExecutionContext.global )
```
Comment (adamw) [inline docs/backends/fs2.md:19]:
```suggestion
implicit val cs: ContextShift[IO] = IO.contextShift(scala.concurrent.ExecutionContext.global)
```

File: docs/backends/fs2.md [hunk 4]
```diff
@@ -44,19 +65,32 @@ To use, add the following dependency to your project:
...
-```scala
+And some imports:
+```scala mdoc:reset:silent
 import sttp.client.httpclient.fs2.HttpClientFs2Backend
+import cats.effect._
+import sttp.client._
 
-HttpClientFs2Backend().flatMap { implicit backend => ... }
+// an implicit ContextShift in required to create a concurrent instance for `cats.effect.IO`:
+implicit val cs: ContextShift[IO] = IO.contextShift( scala.concurrent.ExecutionContext.global )
```
Comment (adamw) [inline docs/backends/fs2.md:75]:
```suggestion
implicit val cs: ContextShift[IO] = IO.contextShift(scala.concurrent.ExecutionContext.global)
```

File: docs/backends/fs2.md [hunk 5]
```diff
@@ -44,19 +65,32 @@ To use, add the following dependency to your project:
...
-
-```scala
+And some imports:
+```scala mdoc:reset:silent
 import sttp.client.httpclient.fs2.HttpClientFs2Backend
+import cats.effect._
+import sttp.client._
 
-HttpClientFs2Backend().flatMap { implicit backend => ... }
+// an implicit ContextShift in required to create a concurrent instance for `cats.effect.IO`:
```
Comment (adamw) [inline docs/backends/fs2.md:74]:
```suggestion
// an implicit ContextShift in required to create a concurrent instance for `cats.effect.IO`,
// as well as a Blocker instance. Note that you'll probably want to use a different thread
// pool for blocking.
```

File: docs/backends/http4s.md [hunk 6]
```diff
@@ -6,18 +6,33 @@ This backend is based on [http4s](https://http4s.org) (blaze client) and is **as
...
-To create the backend, you'll need to provide a `cats.effect.Blocker` instance, which should be passed as a parameter, when defining the backend:
-
-```scala
+Add some imports as well:
+```scala mdoc:silent
+import cats.effect._
 import sttp.client.http4s._
+import scala.concurrent._
+
+// an implicit ContextShift in required to create a concurrent instance for `cats.effect.IO`:
```
Comment (adamw) [inline docs/backends/http4s.md:15]: same as above

File: docs/backends/http4s.md [hunk 7]
```diff
@@ -6,18 +6,33 @@ This backend is based on [http4s](https://http4s.org) (blaze client) and is **as
...
+implicit val cs: ContextShift[IO] = IO.contextShift(ExecutionContext.global)
+val blocker: cats.effect.Blocker = Blocker.liftExecutionContext(ExecutionContext.global)
+```
 
-val blocker: cats.effect.Blocker = ...
+To create the backend, you'll need to provide a `cats.effect.Blocker` instance, which should be passed as a parameter, when defining the backend:
```
Comment (adamw) [inline docs/backends/http4s.md:20]: given the comment above, I think this can be removed

File: docs/conf/proxy.md [hunk 8]
```diff
@@ -12,20 +12,20 @@ Settings are loaded **in given order** and the **first existing value** is being
 
 Otherwise, proxy values can be specified manually when creating a backend:
 
-```scala
+```scala mdoc:compile-only
 import sttp.client._
 
-implicit val backend = HttpURLConnectionBackend(
-  options = SttpBackendOptions.httpProxy("some.host", 8080))
+implicit val backend = HttpURLConnectionBackend(options = SttpBackendOptions.httpProxy("some.host", 8080))
```
Comment (adamw) [inline docs/conf/proxy.md:18]: I'd keep the linebreak so that the rendered code doesn't need to be side-scrolled

File: docs/examples.md [hunk 9]
```diff
@@ -86,17 +87,17 @@ libraryDependencies ++= List(
...
 import sttp.client._
 import sttp.client.circe._
 import sttp.client.asynchttpclient.zio._
 import io.circe.generic.auto._
-import zio._
+import zio.{App => ZApp, _}
 import zio.console.Console
 
-object GetAndParseJsonZioCirce extends App {
+object GetAndParseJsonZioCirce extends ZApp {
```
Comment (adamw) [inline docs/examples.md:98]: I'd use `zio.App` directly

Comment (adamw) [reply docs/examples.md:98]: These files are copy-pasted from files in the `examples` project. Maybe we should drop them and just leave links, or embed?

File: docs/json.md [hunk 10]
```diff
@@ -4,6 +4,12 @@ Adding support for JSON (or other format) bodies in requests/responses is a matt
 
 Each integration is available as an import, which brings the implicit `BodySerializer`s and `asJson` methods into scope. Alternatively, these values are grouped intro traits (e.g. `sttp.client.circe.SttpCirceApi`), which can be extended to group multiple integrations in one object, and thus reduce the number of necessary imports.
 
+Following data class will be used through the next few examples:
+```scala mdoc
+case class Payload(data: String)
```
Comment (adamw) [inline docs/json.md:9]: Maybe we could make this more symmetric, e.g. `RequestPayload` and `ResponsePayload`?

File: docs/requests/basics.md [hunk 11]
```diff
@@ -35,16 +35,16 @@ A request definition can be created without knowing how it will be sent. But to
 
 To invoke the `send()` method on a request description, an implicit value of type `SttpBackend` needs to be in scope:
 
-```scala
+```scala mdoc:compile-only
 implicit val backend = HttpURLConnectionBackend()
-val response: Response[String] = request.send()
+val response = request.send()
```
Comment (adamw) [inline docs/requests/basics.md:40]: can't we keep the type annotation?

File: docs/requests/basics.md [hunk 12]
```diff
@@ -68,16 +68,10 @@ Both of these requests will by default read the response body into a UTF-8 `Stri
 
 sttp comes with builtin request to curl converter. To convert request to curl invocation use `.toCurl` method.
 
-For example converting given request:
+For example:
 
-```scala
+```scala mdoc
```
Comment (adamw) [inline docs/requests/basics.md:73]: 👍

File: docs/requests/body.md [hunk 13]
```diff
@@ -10,25 +10,33 @@ In its simplest form, the request's body can be set as a `String`. By default, t
...
 To set a binary-data body, the following methods are available:
 
-```scala
-def body(b: Array[Byte])
-def body(b: ByteBuffer)
-def body(b: InputStream)
+```scala mdoc:compile-only
+import sttp.client._
+
+val textAsBytes = "Hello, world!".getBytes
```
Comment (adamw) [inline docs/requests/body.md:32]: I think it might be clearer to have:
```
val bytes: Array[Byte] = ???
basicRequest.body(bytes)

val byteBuffer: java.nio.ByteBuffer = ???
basicRequest.bod(byteBuffer)
```

etc. It will be clear (through type annotations) what the signatures are (as we can't show the signatures themselves)

File: docs/responses/basics.md [hunk 14]
```diff
@@ -18,22 +18,30 @@ Response headers are available through the `.headers` property, which gives all
...
-def header(h: String): Option[String]
-def headers(h: String): Seq[String]
+```scala mdoc:silent
+import sttp.client._
+implicit val backend = HttpURLConnectionBackend()
+val request = basicRequest
+    .get(uri"http://endpoint.com/example")
+val response = request.send()
+
+val singleHeader: Option[String] = response.header("Server")
```
Comment (adamw) [inline docs/responses/basics.md:28]: `HeaderNames`? :)

File: docs/responses/basics.md [hunk 15]
```diff
@@ -18,22 +18,30 @@ Response headers are available through the `.headers` property, which gives all
...
 ```
 
 Finally, it's possible to parse the response cookies into a sequence of the `Cookie` case class:
 
-```scala
-def cookies: Seq[Cookie]
+```scala mdoc:compile-only
+import sttp.model._
+
+val cookies: Seq[Cookie] = ???
```
Comment (adamw) [inline docs/responses/basics.md:44]:
```suggestion
val cookies: Seq[Cookie] = response.cookies
```

File: docs/testing.md [hunk 16]
```diff
@@ -81,8 +93,8 @@ val response = basicRequest.get(uri"http://example.org").send()
 
 You can define consecutive raw responses that will be served:
 
-```scala
-implicit val testingBackend = SttpBackendStub.synchronous.whenAnyRequest
+```scala mdoc:compile-only
+implicit val testingBackend:SttpBackendStub[Identity, Nothing, NothingT] = SttpBackendStub.synchronous.whenAnyRequest
```
Comment (adamw) [inline docs/testing.md:97]: I think we'll need a linebreak here

**Comment by adamw** [comment]
`bla.sc`? :)

**Comment by adamw** [comment]
Looks great! :)

File: docs/examples.md (hunk 9 above)
Comment (ghostbuster91) [reply docs/examples.md:98]: What do you mean by embedding? I am not aware of such feature in mdoc

**Comment by ghostbuster91** [comment]
Blocked by: https://github.com/softwaremill/sbt-softwaremill/pull/61

File: .java-version [hunk 17]
```diff
@@ -0,0 +1 @@
+14.0.1
```
Comment (adamw) [inline .java-version:1]: 14? and why is that needed?

File: build.sbt [hunk 18]
```diff
@@ -713,20 +713,62 @@ lazy val examples = (projectMatrix in file("examples"))
 
 val compileDocs: TaskKey[Unit] = taskKey[Unit]("Compiles docs module throwing away its output")
 compileDocs := {
-  (docs.jvm(scala2_12) / mdoc).toTask(" --out target/sttp-docs").value
+  (docs.jvm(scala2_13) / mdoc).toTask(" --out target/sttp-docs").value
 }
 
+val jeagerClientVersion = "1.0.0"
+val braveOpentracingVersion = "0.34.2"
+val zipkinSenderOkHttpVersion = "2.15.0"
```
Comment (adamw) [inline build.sbt:721]: we have a whole section with versions in the build file, maybe we could put it there?

File: build.sbt [hunk 19]
```diff
@@ -713,20 +713,62 @@ lazy val examples = (projectMatrix in file("examples"))
...
-    mdocOut := file("generated-docs/out")
+    mdocOut := file("generated-docs/out"),
+    publishArtifact := false,
+    name := "docs",
+    libraryDependencies ++= Seq(
+      "org.json4s" %% "json4s-native" % json4sVersion,
+      "io.circe" %% "circe-generic" % "0.12.1",
+      "commons-io" % "commons-io" % "2.7",
+      "io.github.resilience4j" % "resilience4j-circuitbreaker" % "1.5.0",
+      "io.github.resilience4j" % "resilience4j-ratelimiter" % "1.5.0",
```
Comment (adamw) [inline build.sbt:743]: extract version

File: docs/backends/fs2.md [hunk 20]
```diff
@@ -19,44 +28,71 @@ Next you'll need to define a backend instance as an implicit value. This can be
...
-
-```scala
+And some imports:
+```scala mdoc:reset:silent
 import sttp.client.httpclient.fs2.HttpClientFs2Backend
+import cats.effect._
+import sttp.client._
 
-HttpClientFs2Backend().flatMap { implicit backend => ... }
+// an implicit ContextShift is required to create a concurrent instance for `cats.effect.IO`,
```
Comment (adamw) [inline docs/backends/fs2.md:74]: you probably mean `cats.effect.Concurrent`? this looks like an adjective, not as a type name. Same for `Blocker`

File: docs/backends/wrappers/opentracing.md [hunk 21]
```diff
@@ -45,16 +47,16 @@ Using with [jaeger](https://www.jaegertracing.io/) tracing
 Add following dependency:
 
 ```
-libraryDependencies += "io.jaegertracing" % "jaeger-client" % "1.0.0"
+libraryDependencies += "io.jaegertracing" % "jaeger-client" % "@JEAGER_CLIENT_VERSION@"
```
Comment (adamw) [inline docs/backends/wrappers/opentracing.md:50]: 👍

File: generated-docs/out/backends/akka.md [hunk 22]
```diff
@@ -3,7 +3,7 @@
 This backend is based on [akka-http](http://doc.akka.io/docs/akka-http/current/scala/http/). To use, add the following dependency to your project:
 
 ```
-"com.softwaremill.sttp.client" %% "akka-http-backend" % "2.2.0"
+"com.softwaremill.sttp.client" %% "akka-http-backend" % "2.2.2-SNAPSHOT"
```
Comment (adamw) [inline generated-docs/out/backends/akka.md:6]: %VERSION?

File: .java-version (hunk 17 above)
Comment (ghostbuster91) [reply .java-version:1]: I have no idea where it came from

File: build.sbt (hunk 18 above)
Comment (ghostbuster91) [reply build.sbt:721]: I put them here since these aren't our dependencies, but they are only used for docs purpose. Since I don't have very strong opinion on that matter I will move them there.

File: build.sbt (hunk 19 above)
Comment (ghostbuster91) [reply build.sbt:743]: Any particular reason for doing that? Not all versions are extracted e.g. 
L431:
```
      "org.asynchttpclient" % "async-http-client" % "2.12.1"
```

I don't see any gains from that, only additional noise.

File: docs/backends/fs2.md [hunk 23]
```diff
@@ -7,7 +7,17 @@ The [fs2](https://github.com/functional-streams-for-scala/fs2) backend is **asyn
...
-"com.softwaremill.sttp.client" %% "async-http-client-backend-fs2" % "2.2.1"
+"com.softwaremill.sttp.client" %% "async-http-client-backend-fs2" % "@VERSION@"
+```
+And some imports:
+```scala mdoc:silent
+import sttp.client.asynchttpclient.fs2.AsyncHttpClientFs2Backend
+import cats.effect._
+import sttp.client._
+
+// an implicit `cats.effect.ContextShift` in required to create an instance of `cats.effectConcurrent`
```
Comment (adamw) [inline docs/backends/fs2.md:18]: typo ;-)
